proxy/api/src/http/project/request.rs

```diff
@@ -0,0 +1,170 @@
+//! Endpoints for project search requests.
+
+use warp::{filters::BoxedFilter, path, Filter, Rejection, Reply};
+
+use crate::{context, http};
+
+/// Combination of all routes.
+pub fn filters(ctx: context::Context) -> BoxedFilter<(impl Reply,)> {
+    cancel_filter(ctx.clone())
+        .or(create_filter(ctx.clone()))
+        .or(list_filter(ctx))
+        .boxed()
+}
+
+/// `DELETE /<urn>`
+fn cancel_filter(
+    ctx: context::Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    http::with_context(ctx)
+        .and(warp::delete())
+        .and(path::param::<coco::Urn>())
+        .and(path::end())
+        .and_then(handler::cancel)
+}
+
+/// `PUT /<urn>`
+fn create_filter(
+    ctx: context::Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    http::with_context(ctx)
+        .and(warp::put())
+        .and(path::param::<coco::Urn>())
+        .and(path::end())
+        .and_then(handler::create)
+}
+
+/// `GET /`
+fn list_filter(
+    ctx: context::Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    http::with_context(ctx)
+        .and(warp::get())
+        .and(path::end())
+        .and_then(handler::list)
+}
+
+/// Request handlers for initiating searches for projects on the network.
+mod handler {
```

Comment:
How would feel about having `project/request/handler.rs`? I've always found it a bit annoying to have these `mod`s in the same file and then I mess up where I import things.

Reply:
It's set up that way because it's the fullfilment of a single chain in a single file. The overhead we had to manage with this is the visibilty `pub`s and import `use` management, which I don't think is worth it.

Reply:
If we change it should be addressed in the entire `http` module.